### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by JayGreeeen**
I've had to swap the client and housekeepingMetadata around in the `MetadataCleaner` so that we can use the `@TimedTaggable` annotation and assert the tag in the tests. Ideally we will remove the client from the method later when we change how it is created.

### CODE REVIEW

File: beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java [hunk 1]
```diff
@@ -48,6 +49,7 @@
 @ContextConfiguration(classes = { TestApplication.class },
   loader = AnnotationConfigContextLoader.class)
 @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+@TestPropertySource(properties = {"spring.jpa.hibernate.ddl-auto=update"})
```
Comment (nvitucci) [inline beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java:52]: Why is this needed here?

Comment (JayGreeeen) [reply beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java:52]: if its not in here we get a very chatty WARN log; `GenerationTarget encountered exception accepting command : Error executing DDL "drop table housekeeping_metadata" via JDBC Statement`.

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java [hunk 2]
```diff
@@ -252,16 +253,21 @@ public void dryRunDontDropPartitionedTable() throws Exception {
   }
 
   @Test
-  public void dryRunMetrics() throws SQLException, TException {
-    hiveTestUtils.createTable(UNPARTITIONED_TABLE_PATH, TABLE_NAME_VALUE, false);
-    amazonS3.putObject(BUCKET, UNPARTITIONED_TABLE_OBJECT_KEY, "");
+  public void dryRunMetrics() throws Exception {
+    Table table = hiveTestUtils.createTable(PARTITIONED_TABLE_PATH, TABLE_NAME_VALUE, true);
```
Comment (nvitucci) [inline beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java:257]: Why has the table become partitioned in this test?

Comment (JayGreeeen) [reply beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java:257]: I changed it so I can check all of the tags are used. For an unpartitioned table it wouldnt use the tag `hive-partition-deleted`, so instead of adding another test for that I changed the table type.

Comment (nvitucci) [reply beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java:257]: I see. Would it be possible to just add it instead (e.g. `dryRunMetricsPartitioned`), so the unpartitioned table still gets tested? Or would it be a waste?

File: beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java (hunk 1 above)
Comment (nvitucci) [reply beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java:52]: I think it's fine to do it since this is just a test, but out of curiosity: what's the full error?

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java (hunk 2 above)
Comment (JayGreeeen) [reply beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java:257]: can do but it would be testing the same tags - 1, so theres not much point imo. The table still gets tested, this is just specifically for the metrics.

File: beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java (hunk 1 above)
Comment (JayGreeeen) [reply beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java:52]: It's a hibernate issue. Seems that the db that gets created is derby and not sql, and it tries to drop the table before creating it, but it doesnt exist so throws the warning. 
`WARN  org.hibernate.tool.schema.internal.ExceptionHandlerLoggedImpl:27 - GenerationTarget encountered exception accepting command : Error executing DDL "drop table housekeeping_metadata" via JDBC Statement
org.hibernate.tool.schema.spi.CommandAcceptanceException: Error executing DDL "drop table housekeeping_metadata" via JDBC Statement
        at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.accept(GenerationTargetToDatabase.java:67)`

File: beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java (hunk 2 above)
Comment (nvitucci) [reply beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperDryRunMetadataCleanupIntegrationTest.java:257]: Ok then, we can leave it this way.

File: beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java (hunk 1 above)
Comment (nvitucci) [reply beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java:52]: Interesting, because for Derby the default for `spring.jpa.hibernate.ddl-auto` should be `create-drop`, which should work. Maybe it's overwritten somewhere else.

Comment (nvitucci) [reply beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporterTest.java:52]: That's odd. Also, why is it referencing the `housekeeping_metadata` table?

**Review by nvitucci — APPROVED** [review]
Looks good.

**Review by massdosage — APPROVED** [review]
